lib/iris/experimental/regrid.py

```diff
@@ -953,10 +953,10 @@ def _src_align_and_flatten(coord):
         min_sx, min_tx = np.min(sx.points), np.min(tx.points)
         if min_sx < 0 and min_tx >= 0:
             indices = np.where(sx_points < 0)
-            sx_points[indices] += modulus
+            sx_points[indices] = sx_points[indices] + modulus
```

Comment:
In the cases where this change fires we are now going to change (i.e. upcast) the dtype of `sx_points`, yes? Is this desirable behaviour, even if it is unavoidable?

Alternatively, was the previous behaviour even correct? Or was it upcasting anyway and just not raising an exception when it did so?